src/python/WMCore/ReqMgr/DataStructs/Request.py

```diff
@@ -92,15 +92,32 @@ def _replace_cloned_args(clone_args, user_args):
             clone_args[prop] = user_args[prop]
     return
     
-def initialize_clone(request_args, reqmgr_db_service):
+def initialize_clone(request_args, originalArgs, argsDefinition):
     """
     Initialize a Clone arguments by inheriting and overwriting argument from OriginalRequest
+
+    :param request_args: user-provided dictionary with override arguments
+    :param originalArgs: original arguments retrieved for the workflow being cloned
+    :param argsDefinition: arguments definition according to the workflow type being cloned
+    :return: dictionary with original args filtered out, as per the spec definition. And on
+     top of that, user arguments added/replaced in the dictionary.
     """
-    requests = reqmgr_db_service.getRequestByNames(request_args["OriginalRequestName"])
-    clone_args = requests.values()[0]
-    # TODO: need to validate new overwrite request_args here since it will skip the clone_args validtaiton
+    clone_args = {}
+    requestType = originalArgs.get("RequestType")
+    for k, v in originalArgs.iteritems():
+        if k in argsDefinition:
+            clone_args[k] = v
+        elif requestType == "ReReco" and k.startswith("Skim"):
+            clone_args[k] = v
+        elif requestType == "StepChain" and k.startswith("Step"):
+            clone_args[k] = v
+        elif requestType == "TaskChain" and k.startswith("Task"):
+            clone_args[k] = v
+        elif requestType == "Resubmission" and k[:4] in ["Skim", "Step", "Task"]:
+            clone_args[k] = v
```

Comment:
Because I wanted to copy only creation & defined arguments, nothing else. The drawback is that we need to take into account those floating arguments from the chained/Rereco specs